### PR TITLE
ticdc: add 4 docs for data replication scenarios

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Suggestion from Allen Gao:
建议增加一个数据集成 一级目录来记录和TiCDC 相关的信息。内容上可以参考 migrate 部分。
- 数据集成 overview: 数据集成的基本信息和 TiCDC 工具介绍。
-  数据集成 scenarios：
    - 与 Confluent Cloud 进行数据集成
    - 与 Apache Kafka 和 Apache Flink 进行数据集成
    
在数据迁移目录下增加《从 TiDB 同步到 MySQL 兼容数据库》

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -114,6 +114,13 @@
     - [下游存在更多列的迁移场景](/migrate-with-more-columns-downstream.md)
     - [如何根据类型或 DDL 内容过滤 binlog 事件](/filter-binlog-event.md)
     - [如何通过 SQL 表达式过滤 DML binlog 事件](/filter-dml-event.md)
+- 数据同步
```
Comment (lilin90) [inline TOC.md:117]:
```suggestion
- 数据复制
```

File: TOC.md [hunk 2]
```diff
@@ -114,6 +114,13 @@
...
+  - [数据同步综述](/replicatation-overview.md)
```
Comment (lilin90) [inline TOC.md:118]:
```suggestion
  - [数据复制综述](/replication-overview.md)
```

File: TOC.md [hunk 3]
```diff
@@ -114,6 +114,13 @@
...
+  - [同步工具](/replication-tools.md)
```
Comment (lilin90) [inline TOC.md:119]:
```suggestion
  - [复制工具](/replication-tool.md)
```

File: migrate-from-tidb-to-mysql.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+1. 搭建环境
+2. 迁移全量数据
+3. 迁移增量数据
+4. 平滑切换业务
+
+## 第 1 步：搭建环境
+
+1. 部署上游 TiDB 集群。
+
+    使用 tiup playground 快速部署上下游测试集群。更多部署信息，请参考 [tiup 官方文档](//tiup/tiup-cluster.md)。
```
Comment (shichun-0415) [inline migrate-from-tidb-to-mysql.md:20]:
```suggestion
    使用 tiup playground 快速部署上下游测试集群。更多部署信息，请参考 [tiup 官方文档](/tiup/tiup-cluster.md)。
```

File: migrate-from-tidb-to-mysql.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    tiup bench tpcc -H 127.0.0.1 -P 4000 -D tpcc --warehouses 4 prepare
+    tiup bench tpcc -H 127.0.0.1 -P 4000 -D tpcc --warehouses 4 run --time 300s
+    ```
+
+    关于 go-tpc 的更多详细内容，可以参考[如何对 TiDB 进行 TPC-C 测试](/benchmark/benchmark-tidb-using-tpcc.md)。
+
+
+## 第 2 步：迁移全量数据
+
+搭建好测试环境后，可以使用 [Dumpling](dumpling-overview.md) 工具导出上游集群的全量数据。
```
Comment (shichun-0415) [inline migrate-from-tidb-to-mysql.md:55]:
```suggestion

## 第 2 步：迁移全量数据

搭建好测试环境后，可以使用 [Dumpling](/dumpling-overview.md) 工具导出上游集群的全量数据。
```

File: migrate-from-tidb-to-mysql.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+     - --pd：实际的上游集群的地址
+     - --sink-uri：同步任务下游的地址
+     - --changefeed-id：同步任务的 ID，格式需要符合正则表达式 ^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$
+     - --start-ts：TiCDC 同步的起点，需要设置为实际的备份时间点（也就是第二章「备份」小节提到的 BackupTS）
+
+    更多关于 changefeed 的配置，请参考[同步任务配置文件描述](/ticdc/manage-ticdc.md#同步任务配置文件描述)。
+
+3. 重新开启 GC。
+
+    TiCDC 可以保证 GC 只回收已经同步的历史数据。因此，创建完从上游到下游集群的 changefeed 之后，就可以执行如下命令恢复集群的垃圾回收功能。详情请参考 [TiCDC GC safepoint 的完整行为](/ticdc/troubleshoot-ticdc.md#ticdc-gc-safepoint-的完整行为是什么)。
```
Comment (shichun-0415) [inline migrate-from-tidb-to-mysql.md:173]:
```suggestion
    TiCDC 可以保证 GC 只回收已经同步的历史数据。因此，创建完从上游到下游集群的 changefeed 之后，就可以执行如下命令恢复集群的垃圾回收功能。详情请参考 [TiCDC GC safepoint 的完整行为](/ticdc/ticdc-faq.md#ticdc-gc-safepoint-的完整行为是什么)。
```

File: replicate-from-tidb-to-confluent.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
+---
+title:  与 Confluent Cloud 进行数据集成
+summary: 了解如何使用 TiCDC 从 TiDB 同步数据至 Confluent Cloud。
```
Comment (shichun-0415) [inline replicate-from-tidb-to-confluent.md:3]:
```suggestion
summary: 了解如何使用 TiCDC 从 TiDB 集成数据至 Confluent Cloud。
```

File: replicate-from-tidb-to-confluent.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+aliases: ['/zh/tidb/dev/integrate-confluent-using-ticdc/']
+---
+
+#  与 Confluent Cloud 进行数据集成
+
+Confluent 是一个兼容 Apache Kafka 的数据流平台，能够访问、存储和管理连续的实时流数据，具备丰富的数据集成能力。自 v6.1.0 开始，TiCDC 支持将增量变更数据以 Avro 格式输出到 Confluent。本文档介绍如何使用 TiCDC 将 TiDB 的增量数据同步到 Confluent Cloud，并借助 Confluent Cloud 的能力最终将数据分别同步到 ksqlDB、Snowflake、SQL Server。主要内容包括：
```
Comment (shichun-0415) [inline replicate-from-tidb-to-confluent.md:9]:
```suggestion
Confluent 是一个兼容 Apache Kafka 的数据流平台，能够访问、存储和管理连续的实时流数据，具备丰富的数据集成能力。自 v6.1.0 开始，TiCDC 支持将增量变更数据以 Avro 格式输出到 Confluent。本文档介绍如何使用 TiCDC 将 TiDB 的增量数据集成到 Confluent Cloud，并借助 Confluent Cloud 的能力最终将数据分别集成到 ksqlDB、Snowflake、SQL Server。主要内容包括：
```

File: replicate-from-tidb-to-kafka-flink.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
+﻿---
+title: 从 TiDB 同步数据至 Apache Kafka 和 Apache Flink
+summary: 了解如何使用 TiCDC 从 TiDB 同步数据至 Apache Kafka 和 Apache Flink。
+aliases: ['/zh/tidb/dev/replicate-data-to-kafka/']
+---
+
+# 与 Apache Kafka 和 Apache Flink 进行数据集成
+
+本文档介绍如何使用 [TiCDC](/ticdc/ticdc-overview.md) 将 TiDB 的数据同步到 Apache Kafka。主要包含以下内容：
```
Comment (shichun-0415) [inline replicate-from-tidb-to-kafka-flink.md:9]:
```suggestion
title: 从 TiDB 集成数据至 Apache Kafka 和 Apache Flink
summary: 了解如何使用 TiCDC 从 TiDB 集成数据至 Apache Kafka 和 Apache Flink。
aliases: ['/zh/tidb/dev/replicate-data-to-kafka/']
---

# 与 Apache Kafka 和 Apache Flink 进行数据集成

本文档介绍如何使用 [TiCDC](/ticdc/ticdc-overview.md) 将 TiDB 的数据集成到 Apache Kafka。主要包含以下内容：
```

File: replication-overview.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
+---
+title: 数据集成场景综述
+summary: 了解使用 TiCDC 进行数据集成的具体场景。
+---
+
+# 数据集成场景综述
+
+本文档总体介绍可用于 TiDB 的数据同步方案。
```
Comment (shichun-0415) [inline replication-overview.md:8]:
```suggestion
本文档总体介绍可用于 TiDB 的数据集成方案。
```

File: replication-overview.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+## 与 Confluent Cloud 进行数据集成
+
+您可以使用 TiCDC 将 TiDB 的增量数据同步到 Confluent Cloud，并借助 Confluent Cloud 的能力最终将数据分别同步到 ksqlDB、Snowflake、SQL Server。
+
+[与 Confluent Cloud 进行数据集成](/replicate-from-tidb-to-confluent.md)
+
+## 与 Apache Kafka 和 Apache Flink 进行数据集成
+
+您可以使用 TiCDC 将 TiDB 的增量数据同步到 Apache Kafka，并使用 Apache Flink 消费 Kafka 中的数据。
```
Comment (shichun-0415) [inline replication-overview.md:18]:
```suggestion
您可以使用 TiCDC 将 TiDB 的增量数据集成到 Confluent Cloud，并借助 Confluent Cloud 的能力最终将数据分别集成到 ksqlDB、Snowflake、SQL Server。

[与 Confluent Cloud 进行数据集成](/replicate-from-tidb-to-confluent.md)

## 与 Apache Kafka 和 Apache Flink 进行数据集成

您可以使用 TiCDC 将 TiDB 的增量数据集成到 Apache Kafka，并使用 Apache Flink 消费 Kafka 中的数据。
```

File: replicate-from-tidb-to-kafka-flink.md [hunk 12]
```diff
@@ -0,0 +1,165 @@
+﻿---
+title: 从 TiDB 同步数据至 Apache Kafka 和 Apache Flink
```
Comment (leoppro) [inline replicate-from-tidb-to-kafka-flink.md:2]:
```suggestion
title: 与 Apache Kafka 和 Apache Flink 进行数据集成
```

**Review by sdojjy — APPROVED** [review]

File: TOC.md [hunk 13]
```diff
@@ -108,12 +108,17 @@
     - [从 CSV 文件迁移数据到 TiDB](/migrate-from-csv-files-to-tidb.md)
     - [从 SQL 文件迁移数据到 TiDB](/migrate-from-sql-files-to-tidb.md)
     - [从 TiDB 集群迁移数据至另一 TiDB 集群](/migrate-from-tidb-to-tidb.md)
-    - [从 TiDB 同步数据至 Apache Kafka](/replicate-data-to-kafka.md)
+    - [从 TiDB 集群迁移数据至 MySQL 兼容的数据库](/migrate-from-tidb-to-mysql.md)
```
Comment (lilin90) [inline TOC.md:111]:
```suggestion
    - [从 TiDB 集群迁移数据至兼容 MySQL 的数据库](/migrate-from-tidb-to-mysql.md)
```

File: integrate-with-kafka-flink.md [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+
+        ```shell
+        Create changefeed successfully!
+        ID: kafka-changefeed
+        Info: {... changfeed info json struct ...}
+        ```
+
+    - 如果命令长时间没有返回，你需要检查当前执行命令所在服务器到 sink-uri 中指定的 Kafka 机器的网络可达性，保证二者之间的网络连接正常。
+
+    生产环境下 Kafka 集群通常有多个 broker 节点，你可以在 sink-uri 中配置多个 broker 的访问地址，这有助于提升 changefeed 到 Kafka 集群访问的稳定性，当部分被配置的 Kafka 节点故障的时候，changefeed 依旧可以正常工作。假设 Kafka 集群中有 3 个 broker 节点，地址分别为 127.0.0.1:9092 / 127.0.0.2:9092 / 127.0.0.3:9092，可以参考如下 sink-uri 创建 changefeed:
```
Comment (lilin90) [inline integrate-with-kafka-flink.md:74]:
```suggestion
    生产环境下 Kafka 集群通常有多个 broker 节点，你可以在 sink-uri 中配置多个 broker 的访问地址，这有助于提升 changefeed 到 Kafka 集群访问的稳定性，当部分被配置的 Kafka 节点故障的时候，changefeed 依旧可以正常工作。假设 Kafka 集群中有 3 个 broker 节点，地址分别为 127.0.0.1:9092 / 127.0.0.2:9092 / 127.0.0.3:9092，可以参考如下 sink-uri 创建 changefeed：
```

File: integrate-with-kafka-flink.md [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+    ```shell
+    tiup ctl:v6.1.0 cdc changefeed create --pd="http://127.0.0.1:2379" --sink-uri="kafka://127.0.0.1:9092,127.0.0.2:9092,127.0.0.3:9092/kafka-topic-name?protocol=canal-json&partition-num=3&replication-factor=1&max-message-bytes=1048576" --config="changefeed.conf"
+    ```
+
+3. changefeed 创建成功后，执行如下命令，查看 changefeed 的状态:
```
Comment (lilin90) [inline integrate-with-kafka-flink.md:80]:
```suggestion
3. Changefeed 创建成功后，执行如下命令，查看 changefeed 的状态：
```

File: migrate-from-tidb-to-mysql.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
+---
+title: 从 TiDB 集群迁移数据至 MySQL 兼容的数据库
+summary: 了解如何将数据从 TiDB 集群迁移至 MySQL 兼容数据库
+---
+
+# 从 TiDB 集群迁移数据至 MySQL 兼容的数据库
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:6]:
```suggestion
# 从 TiDB 集群迁移数据至兼容 MySQL 的数据库
```

File: migrate-from-tidb-to-mysql.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
+---
+title: 从 TiDB 集群迁移数据至 MySQL 兼容的数据库
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:2]:
```suggestion
title: 从 TiDB 集群迁移数据至兼容 MySQL 的数据库
```

File: migrate-from-tidb-to-mysql.md [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+summary: 了解如何将数据从 TiDB 集群迁移至 MySQL 兼容数据库
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:3]:
```suggestion
summary: 了解如何将数据从 TiDB 集群迁移至与 MySQL 兼容的数据库。
```

File: migrate-from-tidb-to-mysql.md [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+---
+
+# 从 TiDB 集群迁移数据至 MySQL 兼容的数据库
+
+本文档介绍如何将数据从 TiDB 集群迁移至 MySQL 兼容的数据库，如 Aurora、MySQL、MariaDB 等。本文将模拟整个迁移过程，具体包括以下四个步骤：
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:8]:
```suggestion
本文档介绍如何将数据从 TiDB 集群迁移至兼容 MySQL 的数据库，如 Aurora、MySQL、MariaDB 等。本文将模拟整个迁移过程，具体包括以下四个步骤：
```

File: migrate-from-tidb-to-mysql.md [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+    tiup bench tpcc -H 127.0.0.1 -P 4000 -D tpcc --warehouses 4 run --time 300s
+    ```
+
+    关于 go-tpc 的更多详细内容，可以参考[如何对 TiDB 进行 TPC-C 测试](/benchmark/benchmark-tidb-using-tpcc.md)。
+
+## 第 2 步：迁移全量数据
+
+搭建好测试环境后，可以使用 [Dumpling](/dumpling-overview.md) 工具导出上游集群的全量数据。
+
+1. 关闭 GC。
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:53]:
```suggestion
1. 关闭 GC (Garbage Collection)。
```

File: migrate-from-tidb-to-mysql.md [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
+    | @@global.tidb_gc_enable |
+    +-------------------------+
+    |                       0 |
+    +-------------------------+
+    1 row in set (0.00 sec)
+    ```
+
+    > **注意：**
+    >
+    > 在生产集群中，关闭 GC 机制和备份操作会一定程度上降低集群的读性能，建议在业务低峰期进行备份，并设置合适的 RATE_LIMIT 限制备份操作对线上业务的影响。
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:71]:
```suggestion
    > 在生产集群中，关闭 GC 机制和备份操作会一定程度上降低集群的读性能，建议在业务低峰期进行备份，并设置合适的 `RATE_LIMIT` 限制备份操作对线上业务的影响。
```

File: migrate-from-tidb-to-mysql.md [hunk 22]
```diff
@@ -0,0 +1,200 @@
...
+    ```shell
+    tiup ctl:v6.1.0 cdc changefeed create --pd=http://127.0.0.1:2379 --sink-uri="mysql://root:@127.0.0.1:3306" --changefeed-id="upstream-to-downstream" --start-ts="434217889191428107"
+    ```
+
+    以上命令中：
+
+     - --pd：实际的上游集群的地址
+     - --sink-uri：同步任务下游的地址
+     - --changefeed-id：同步任务的 ID，格式需要符合正则表达式 ^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$
+     - --start-ts：TiCDC 同步的起点，需要设置为实际的备份时间点（也就是第二章「备份」小节提到的 BackupTS）
```
Comment (lilin90) [inline migrate-from-tidb-to-mysql.md:155]:
```suggestion
     - `--pd`：实际的上游集群的地址
     - `--sink-uri`：同步任务下游的地址
     - `--changefeed-id`：同步任务的 ID，格式需要符合正则表达式 `^[a-zA-Z0-9]+(\-[a-zA-Z0-9]+)*$`
     - `--start-ts`：TiCDC 同步的起点，需要设置为实际的备份时间点（也就是第二章「备份」小节提到的 BackupTS）
```

Comment (lilin90) [reply migrate-from-tidb-to-mysql.md:155]: “第二章「备份」”指的是哪里呢？`「」`这个符号不太正式，在正式文本中请尽量少用。

File: replicate-data-to-kafka.md [hunk 23]
```diff
@@ -1,106 +1,168 @@
...
 
-    生产环境下，你需要自行开发 Kafka Consumer 程序，对 Kafka Topic 中的数据进行消费。
\ No newline at end of file
+至此，TiDB 的增量数据变更日志就实时地复制到了 Kafka。下一步，你可以使用 Flink 消费 Kafka 数据。当然，你也可以自行开发适用于业务场景的 Kafka 消费端。
+
+## 第 4 步：配置 Flink 消费 Kafka 数据（可选）
+
+1. 安装 Flink Kafka Connector。
+
+    在 Flink 生态中，Flink Kafka Connector 用于消费 Kafka 中的数据并输出到 Flink 中。Flink Kafka Connector 并不是内建的，因此在 Flink 安装完毕后，还需要将 Flink Kafka Connector 及其依赖项添加到 Flink 安装目录中。下载下列 jar 文件至 Flink 安装目录下的 lib 目录中，如果您已经运行了 Flink 集群，请重启集群以加载新的插件。
```
Comment (lilin90) [inline replicate-data-to-kafka.md:117]:
```suggestion
    在 Flink 生态中，Flink Kafka Connector 用于消费 Kafka 中的数据并输出到 Flink 中。Flink Kafka Connector 并不是内建的，因此在 Flink 安装完毕后，还需要将 Flink Kafka Connector 及其依赖项添加到 Flink 安装目录中。下载下列 jar 文件至 Flink 安装目录下的 lib 目录中，如果你已经运行了 Flink 集群，请重启集群以加载新的插件。
```

File: ticdc/integrate-confluent-using-ticdc.md [hunk 24]
```diff
@@ -1,130 +1,265 @@
 ---
-title: TiDB 集成 Confluent Platform 快速上手指南
-summary: 了解如何使用 TiCDC 将 TiDB 数据流式传输到 Confluent Platform。
+title:  与 Confluent Cloud 进行数据集成
```
Comment (lilin90) [inline ticdc/integrate-confluent-using-ticdc.md:2]:
```suggestion
title: 与 Confluent Cloud 进行数据集成
```

File: replicate-between-primary-and-secondary-clusters.md [hunk 25]
```diff
@@ -247,9 +243,9 @@ summary: 了解如何配置一个 TiDB 集群以及该集群的 TiDB 或 MySQL 
 
     以上命令中：
 
-   * --pd: 实际的上游集群的地址
-   * --sink-uri：同步任务下游的地址
-   * --start-ts: TiCDC 同步的起点，需要设置为实际的备份时间点（也就是[第 2 步：迁移全量数据](#第-2-步迁移全量数据)提到的 BackupTS）
+    - --pd：实际的上游集群的地址
+    - --sink-uri：同步任务下游的地址
+    - --start-ts：TiCDC 同步的起点，需要设置为实际的备份时间点（也就是[第 2 步：迁移全量数据](#第-2-步迁移全量数据)提到的 BackupTS）
```
Comment (lilin90) [inline replicate-between-primary-and-secondary-clusters.md:248]:
```suggestion
    - `--pd`：实际的上游集群的地址
    - `--sink-uri`：同步任务下游的地址
    - `--start-ts`：TiCDC 同步的起点，需要设置为实际的备份时间点（也就是[第 2 步：迁移全量数据](#第-2-步迁移全量数据)提到的 BackupTS）
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**